src/main/java/com/Application/Tracking_System_Application.java

```diff
@@ -9,6 +11,7 @@
 
 @SpringBootApplication
 @ComponentScan(basePackageClasses = EventController.class)
+@ComponentScan(basePackageClasses = EmailSenderService.class)
```

Comment:
Doesn't @SpringBootApplication already do the ComponentScan?

Reply:
It supposes to, but it seems like they do not recognize those classes once I put them into subdirectories, if I delete @ComponentScan, they cannot auto-wire EventController and EmailSenderService anymore.